quickstart/101-key-vault-key/main.tf

```diff
@@ -0,0 +1,62 @@
+resource "random_pet" "rg_name" {
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = random_pet.rg_name.id
+  location = var.resource_group_location
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "random_string" "azurerm_key_vault_name" {
+  length  = 13
+  lower   = true
+  numeric = false
+  special = false
+  upper   = false
+}
+
+resource "azurerm_key_vault" "vault" {
+  name                       = (var.vault_name != "") ? var.vault_name : "vault-${random_string.azurerm_key_vault_name.result}"
+  location                   = azurerm_resource_group.rg.location
+  resource_group_name        = azurerm_resource_group.rg.name
+  tenant_id                  = data.azurerm_client_config.current.tenant_id
+  sku_name                   = var.sku_name
+  soft_delete_retention_days = 7
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = data.azurerm_client_config.current.object_id
+
+    key_permissions    = var.key_permissions
+    secret_permissions = var.secret_permissions
+  }
+}
+
+resource "random_string" "azurerm_key_vault_key_name" {
+  length  = 13
+  lower   = true
+  numeric = false
+  special = false
+  upper   = false
+}
+
+resource "azurerm_key_vault_key" "key" {
+  name = ((var.key_name != "")
+  ? var.key_name : "key-${random_string.azurerm_key_vault_key_name.result}")
+
```

Comment:
Could we simplify it to?:

```hcl
name = coalesce(var.key_name, "key-${random_string.azurerm_key_vault_key_name.result}")
```

Reply:
@lonegunmanb The coalesce fcn is new to me. Thanks!